src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandParserSupport.cs

```diff
@@ -95,23 +95,17 @@ public static HashSet<string> ArgsForBuiltInCommands
                     aliasesForParam.Add(shortVersion);
                 }
 
-                if (string.Equals(parameter.DataType, "choice", StringComparison.OrdinalIgnoreCase))
+                if (parameter is IAllowDefaultIfOptionWithoutValue parameterWithNoValueDefault
+                    && !string.IsNullOrEmpty(parameterWithNoValueDefault.DefaultIfOptionWithoutValue))
```

Comment:
Could empty string be a valid default?

Reply:
For the normal default we don't differentiate between null & empty string, which effectively makes an empty string (regular) default not possible. I think we should treat possible values for this new default the same as the regular default. My thought is that we shouldn't allow empty string defaults, but it would be possible. We should discuss the possibilities.